Source/CarthageKit/Version.swift

```diff
@@ -70,6 +75,14 @@ public struct SemanticVersion: VersionType {
 			}
 		}
 	}
+	
+	/// The same SemanticVersion as self, except that the build metadata is discarded
+	public var discardingBuildMetadata: SemanticVersion {
```

Comment:
Do we even care about the build metadata?

Let's just remove it from `SemanticVersion` entirely. (But allow it in the parsing code.)

Reply:
What about `.exactly`? 

I think it makes no sense to consider build metadata for `.atLeast` (also, no clear ordering is defined with build metadata anyway) and `.compatibleWith` (any other version with same major-minor-patch, without the build metadata, is compatible). 

However I would find if weird if when specifying `exactly 1.0.2+build2`, it would resolve to `1.0.2+build8997`. Is there any case in which you see this as desired?

Reply:
Hmm… that's tricky. I'm not sure this makes sense, so my preference would be to disallow it for now. Maybe we should just not even parse it? I don't know if anyone even really uses this.

Reply:
I'm definitely open to ideas on this, because I'm not really sure what the right answer is.

Reply:
I considered and discarded the following alternatives:

1. discarding build metadata completely. This feels broken, as when I say that I want *exactly* build `1.2.0+b32`, Carthage will instead pick `1.2.0+b8934`.
1. consider build metadata when resolving for `.compatibleWith`. This makes no sense: by definition, all version with the same major-minor-patch are compatible, so a different build metadata should not make a version not compatible.
1. consider build metadata when resolving for `.atLeast`. This goes against SemVer specifications as there is no ordering defined between build versions. From the [specs](https://semver.org/#spec-item-10): 
> Build metadata SHOULD be ignored when determining version precedence. Thus two versions that differ only in the build metadata, have the same precedence.

One alternative I can think of would be to modify the first rule in the following way:

- if required to take `exactly 1.2.0+b32` (i.e. the requirement has build metadata), then only the version with that exact build metadata should be picked.
- if required to take `exactly 1.2.0` (i.e. no build metadata), then it will pick *any* `1.2.0` version even if they have a build metadata, e.g. it would pick `1.2.0+b42`. If there are multiple ones, it would pick the first one matching.

Reply:
We can roll with it as-is and make adjustments down the road if needed.